### PR TITLE
feat(select): select component

### PR DESCRIPTION
**Pull request by AngeliaQ**
## Checklist

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/react-ui/35)

### CODE REVIEW

File: components/select/select-option.tsx [hunk 1]
```diff
@@ -52,20 +54,40 @@ const SelectOption: React.FC<React.PropsWithChildren<SelectOptionProps>> = ({
...
 
   const color = useMemo(() => {
-    if (isDisabled) return theme.palette.accents_4
-    return selected ? theme.palette.foreground : theme.palette.accents_5
+    if (isDisabled) return theme.palette.cNeutral4
+    return selected
+      ? variant === 'text'
+        ? theme.palette.cTheme5
+        : theme.palette.cNeutral8
+      : theme.palette.cNeutral6
```
Comment (yqrashawn) [inline components/select/select-option.tsx]: Better use `if...else` or something more clear here.

File: components/select/select-option.tsx [hunk 2]
```diff
@@ -52,20 +54,40 @@ const SelectOption: React.FC<React.PropsWithChildren<SelectOptionProps>> = ({
...
+  const border = useMemo(() => {
+    if (isDisabled || variant !== 'text') return '0'
+    return selected ? `1px solid ${theme.palette.cTheme5}` : `1px solid ${theme.palette.cNeutral8}`
+  }, [selected, isDisabled, theme.palette])
+
+  const hoverBorder = useMemo(() => {
+    if (isDisabled || isLabel || selected || variant !== 'text') return border
+    return `1px solid rgba(${rgb.r}, ${rgb.g}, ${rgb.b}, 0.12)`
+  }, [selected, isDisabled, theme.palette, isLabel, color])
+
```
Comment (yqrashawn) [inline components/select/select-option.tsx]: Better put all these into a sperate style file.

File: components/select/select.tsx [hunk 3]
```diff
@@ -168,12 +175,12 @@ const Select: React.FC<React.PropsWithChildren<SelectProps>> = ({
             width: ${width};
             overflow: hidden;
             transition: border 0.2s ease 0s, color 0.2s ease-out 0s, box-shadow 0.2s ease 0s;
-            border: 1px solid ${theme.palette.border};
+            border: ${isText ? 'none' : `1px solid ${theme.palette.cNeutral3}`};
```
Comment (yqrashawn) [inline components/select/select.tsx]: Better put this kind of logic into a separate style file

File: components/select/select.tsx [hunk 4]
```diff
@@ -184,11 +191,12 @@ const Select: React.FC<React.PropsWithChildren<SelectProps>> = ({
           }
 
           .select:hover {
-            border-color: ${disabled ? theme.palette.border : theme.palette.foreground};
+            border-color: ${disabled ? theme.palette.cNeutral3 : theme.palette.cTheme5};
```
Comment (yqrashawn) [inline components/select/select.tsx]: Better put this kind of logic into a separate style file

**Review by yqrashawn — CHANGES_REQUESTED** [review]

File: components/select/select-option.tsx (hunk 1 above)
Comment (AngeliaQ) [reply components/select/select-option.tsx]: the code has been moved to style.ts

File: components/select/select-option.tsx (hunk 2 above)
Comment (AngeliaQ) [reply components/select/select-option.tsx]: done

File: components/select/select.tsx (hunk 3 above)
Comment (AngeliaQ) [reply components/select/select.tsx]: done

**Review by 0x74616e67 — DISMISSED** [review]

**Review by yqrashawn — DISMISSED** [review]

**Review by 0x74616e67 — APPROVED** [review]

**Review by yqrashawn — APPROVED** [review]

**Review by 0x74616e67 — APPROVED** [review]

**Pull request merged**